src/main/java/luke/Luke.java

```diff
@@ -0,0 +1,78 @@
+package luke;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.nio.file.FileAlreadyExistsException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Scanner;
+
+public class Luke {
+
+    //Logo created using https://patorjk.com/software/taag/#p=display&f=Varsity&t=Luke
+    private static String logo = "  _____             __             \n"
+            + " |_   _|           [  |  _         \n"
+            + "   | |     __   _   | | / ] .---.  \n"
+            + "   | |   _[  | | |  | '' < / /__\\\\ \n"
+            + "  _| |__/ || \\_/ |, | |`\\ \\| \\__., \n"
+            + " |________|'.__.'_/[__|  \\_]'.__.' ";
+
+    public static void main(String[] args) throws LukeException {
+        Storage storage;
+        UI ui = new UI();
+        ui.greet();
+
+        String wd = System.getProperty("user.dir");
+        Path directoryPath = Paths.get(wd,  "data");
+        Path historyPath = Paths.get(wd, "data", "history.txt");
+        //create directory (if it does not already exist)
+        try {
+            Files.createDirectory(directoryPath);
+        } catch (FileAlreadyExistsException e) {
+            //directory already exists.
+        } catch (IOException e) {
+            System.out.println("Failed to create directory");
+        }
+        //create file (if it does not already exist)
+        try {
+            Files.createFile(historyPath);
+        } catch (FileAlreadyExistsException e) {
+            //file already exists.
+        } catch (IOException e) {
+            System.out.println("Failed to create file");
+            return;
+        }
+
+        //load the file if there is save data (reference: https://www.baeldung.com/java-serialization)
+        //otherwise, create a new History object.
+        File historyFile = new File(String.valueOf(historyPath));
```

Comment:
The process of loading the storage file may be more suitable to be done by the Storage class.